floris/tools/floris_interface.py

```diff
@@ -820,6 +820,42 @@ def get_farm_AEP(self, wd, ws, freq, yaw=None):
             AEP_sum = AEP_sum + self.get_farm_power()*freq[i]*8760
         return AEP_sum
 
+    def change_turbine(self, turb_num_array, turbine_change_dict):
+        """
+        Change turbine properties of given turbines
+
+        Args:
+            turb_num_array (list): list of turbines to apply the change to
+            turbine_change_dict (dict): dictionary of changes to make.  All 
+                key values should be from the JSON turbine/properties set.
+                Any key values not specified will be copied from the original
+                JSON values
+        """
+
+        #TODO This may not work as intended if floris_interface not initialized with JSON
+
+        # First re-read the input file to get the baseline parameters
+        json_dict = self.floris.input_reader._parseJSON(self.input_file)
```

Comment:
Can you use the `input_dict` on the floris object instead of parsing again? Also, you should generally avoid using anything with a leading underscore.

Reply:
That's a good catch, I missed that floris anyway saves that, I'll change it

Reply:
Hmm, it's not clear the resultant dict is saved, it seems to be read in and passed in line 88 of floris.py, I think that's why I thought I needed to read it again...

Reply:
Good point. As an alternative, I propose that this modifies the Turbine objects in place rather than going back to the input file and starting from scratch. All properties defined in the input file exist in the Turbine class and for any that don't have an explicit setter-method we can add one. This also takes care of the situation in your TODO where FLORIS was instantiated from a dictionary instead of an input file. What do you think?

Reply:
I like this idea, but I wasn't sure how to implement it.  Do you have any suggestions, or would like to take a crack at it?